modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/DefaultQuestionsEditor.es.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useEffect, useState} from 'react';
+
+import {stripHTML} from '../utils/utils.es';
+import QuestionsEditor from './QuestionsEditor';
+import TextLengthValidation from './TextLengthValidation.es';
+
+export default ({
+	additionalInformation = '',
+	cleanContent,
+	contentRefFromParent,
+	defaultContent = '',
+	disableActionButton,
+	label = '',
+	question = {locked: false},
+	...props
+}) => {
+	const MIN_CHAR_NUMBER = 15;
+	const [articleBody, setArticleBody] = useState('');
+
+	useEffect(() => {
+		setArticleBody('');
+	}, [cleanContent]);
+
+	useEffect(() => {
+		setArticleBody(defaultContent ? defaultContent : '');
+	}, [defaultContent]);
+
+	useEffect(() => {
+		contentRefFromParent.current = articleBody;
+		disableActionButton(stripHTML(articleBody).length < MIN_CHAR_NUMBER);
+	}, [articleBody, contentRefFromParent, disableActionButton]);
```

Comment:
Instead of having this component react to a prop change and calling an internal setState via useEffect, I think it would be better to use ref's to expose a method from this component. Take a look at my example below of a simple use case of this

```jsx
const ChildEditor = React.forwardRef((props, ref) => {
  const [content, setContent] = React.useState("");

  ref.current = {
    getContent: () => content,
    clearConent: () => setContent(""),
  };

  return <input onChange={(e) => setContent(e.target.value)} value={content} />;
});

const ParentComponent = () => {
  const editorRef = React.useRef(null);

  function handleSubmit() {
    const editorContent = editorRef.current.getContent();

    // do something here to submit this content
  }

  return (
    <div>
      <ChildEditor ref={editorRef} />

      <button onClick={handleSubmit}>Submit</button>
    </div>
  );
};
```

This is an especially common pattern for "Editor-like" components since we want to keep the state changes and re-renders isolated. This also allows us to only grab the content we need at submit, rather than every keystroke

Reply:
If this doesn't make sense at all, I am happy to jump on a call and try to clear it up as well